src/frontend/lib/date.ts

```diff
@@ -1,3 +1,19 @@
+export const SECOND_MS = 1000;
+export const MINUTE_MS = SECOND_MS * 60;
+export const HOUR_MS = MINUTE_MS * 60;
+export const DAY_MS = HOUR_MS * 24;
+const ISO_DATE_REGEXP = /^(?<year>\d{4,})-(?<month>\d{2})-(?<day>\d{2})$/;
+
+export const beginningOfMonthUtc = (date: Readonly<Date>): Date =>
+  new Date(Date.UTC(date.getUTCFullYear(), date.getUTCMonth(), 1));
+
+export const formatIsoUtc = (date: Readonly<Date>): string =>
+  [
+    date.getUTCFullYear().toString().padStart(4, '0'),
+    (date.getUTCMonth() + 1).toString().padStart(2, '0'),
+    date.getDate().toString().padStart(2, '0'),
```

Comment:
I think this should be `date.getUTCDate()`?

Reply:
Maybe a test case needs added to catch this?

Reply:
See #593 and #595.